### DISCUSSION TITLE
GPS coordinates: questions / ideas

### DISCUSSION THREAD

**Discussion by pietricello** [zepp-health/zeppos-forum#98] [General]
Most of our watches has gps built in. I saw several apps for navigation on Zepp OS but i never found something as simply as useful: a simply watch app that shows coordinates (altitude as plus) . Simply: xx.xxxx, yy.yyyy. It could be useful just to report where u are in case (e.g) you lost your phone or battery KO. More than suggestion (i'm not good in programming) my question is : is it impossibile or simply this does not interest anyone? The idea is : some of you guys so good in programming why do not write a simple app like this? Unless there are no API for getting a-gps data i think should be easy to code no? Any answer is appreciated :-)

**Comment by vaas1993** [comment]
After reading the documents of ZeppOS, it seems that there is no open API for accessing GPS positioning data, or I can't find it?

However, even if the relevant API is provided, the reality does not seem to be the same as you think. Maintaining GPS positioning for a long time is very power-consuming, and the solution to obtain positioning at a certain interval may not solve your problem

Sorry, I'm a Chinese developer. The above text is machine translated

**Comment by pietricello** [comment]
Hi thank you for your answer. Guess the true issue is that maybe there is no open API. On the other side my idea is no so much complicated, then, no power-consuming. Not interested in tracking but simple having on demand the gps coordinates, for instance, in the compass application that already is on the watch as some apps on ios/android done. Technically speaking my GTR3 Pro has the gps because when i start a walk it stores the gps data than further are available on zepp app, it means that it knows "where i am". Just to speak ...why this? In my job everyday i ask customer's position for help them. I have to explain "go to maps...keep the finger on the dot...tell me the numbers you read" then i thought "why i have to use the phone if i already wear a gps capable watch?". So nothing of complicated in my head just that 2 coordinates (static in the moment you request as per walk or other activities) on the watch :-)

**Comment by joaquinperaza** [comment]
I'm also interested on having an open location API GPS+altimeter

**Comment by MyoungXUE** [comment]
Hello, Geolocation and Barometer will be available on Zepp OS 3.0. You can follow our Roadmap https://github.com/orgs/zepp-health/discussions/59, please stay tuned

**Reply by joaquinperaza** [reply]
Thanks for this great update! I will be looking forward to test it, do we need to wait until October to test it, or can we start developing apps earlier debugging them
on the simulator?
